### ISSUE TITLE
String literals

### ISSUE THREAD

**Issue by aubreyrjones** [llvm-dcpu16/llvm-dcpu16#90]
My plan is to use native DCPU characters. So, I want 16-bit characters, lower octet character, higher octet control. All of my string functions then will be defined on the word, but lexically work only on the low byte. Should be straightforward. And much faster than trying to convert between 8-bit and 16-bit on the fly.

But, I'm having trouble coming up a "clean" 16-bit character literal.

(Incidentally, I know that y'all are trying to get an 8-bit 'char' type. Please don't be distracted. The DCPU-16 has no facilities for sub-word access. Working with a char to "save" 8 bits is going to result in 3 times the operations as it gets bit twiddled. And there appears to be no other infrastructure expecting 8-bit characters.)

Here's what I had the best luck with so far. But it throws a warning. I hadn't realized that the L string literal was an int. I had thought it was a short. And I have no idea what type sizes y'all have defined for stuff.

```
str_test.c:3:15: warning: incompatible pointer types initializing 'short *' with an 
expression of type 'int [44]' [-Wincompatible-pointer-types]
```

```
static short *wTest = L"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
:wTest
        .short  .L.str

        .section        .rodata.str2.2,"aMS",@progbits,2
        ; .align        1
:.L.str
        .short  97
        .short  97
        .short  97
        .short  97
        .short  97
        .short  97
        .short  97
        .short  97
        .short  97
    .short  97
        .short  97
```

**Comment by krasin** [comment]
Hi Aubrey,

we're waiting for the upcoming changes of DCPU16 from Notch. After that, we'll figure out the way to support string and this issue would be the thread for discussing this thing.

Thanks for filing!

**Comment by ghost** [comment]
This should also work now. Since we have wide chars now.

**Comment by aubreyrjones** [comment]
Are you emitting .asciz as the directive still? (This is what clang was emitting, uh, two days ago when I last rebuilt.)

**Comment by ghost** [comment]
Haven't changed anything in the emitting code yet, so yeah, I think it will still emit that directive. Can't test right now.

**Comment by ghost** [comment]
Fixed by 10790544e9d2441da92ab3593beb185ef2b08dfa
